webhooks/appmesh/virtualnode_validator.go

```diff
@@ -63,6 +71,30 @@ func (v *virtualNodeValidator) enforceFieldsImmutability(vn *appmesh.VirtualNode
 	return nil
 }
 
+func (v *virtualNodeValidator) checkVirtualNodeBackendsForDuplicates(vn *appmesh.VirtualNode) error {
+	backends := vn.Spec.Backends
+	backendMap := make(map[string]bool, len(backends))
+
+	for _, backend := range backends {
+		if backend.VirtualService.VirtualServiceRef != nil {
+			backendNamespacedName := references.ObjectKeyForVirtualServiceReference(vn, *backend.VirtualService.VirtualServiceRef)
+			backendIdentifier := backendNamespacedName.Name + "-" + backendNamespacedName.Namespace
+			if _, ok := backendMap[backendIdentifier]; ok {
+				return errors.Errorf("%s-%s has duplicate VirtualServiceReferences %s", "VirtualNode", vn.Name, backend.VirtualService.VirtualServiceRef.Name)
```

Comment:
nit: better to log the namespace name as well. Will help with easier debugging in large clusters with duplicate names across namespaces

Reply:
Name of the duplicate service should help us find out the duplicates in the VN Spec. I mean since we throw an error only if the namespace is same for both the Virtualservices sharing the same name. Believe we should be ok..